### PR TITLE
Pie Graph Take 2

### PR DESCRIPTION
**Pull request by ChristianMurphy**
#491 has been reverted as it is breaking the builds

### CODE REVIEW

File: application.properties [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #Grails Metadata file
-app.grails.version=2.5.1
+app.grails.version=2.5.0
```
Comment (ChristianMurphy) [inline application.properties]: Grails 2.5.1 contains stability and security fixes we will be using this going forward.
Also both build systems are set to use 2.5.1, this line will cause the build to fail.

File: grails-app/views/content/index.gsp [hunk 2]
```diff
@@ -29,6 +29,8 @@
 		<link id="imgCFMP" rel="prefetch" href="${resource(dir: 'images/content', file: 'knowDimCFMP.png')}">
 
 		<g:javascript src="source/topicDialog.js" defer="defer" />
+		<g:external dir="js/plugins" file="jquery.piegraph.js"/>
+		<g:external dir="js/plugins" file="jquery.canvasjs.min.js"/>
```
Comment (ChristianMurphy) [inline grails-app/views/content/index.gsp]: files in the plugins folder are not check into git.
Bower manages loading in these files.
When adding a new plugin add it to the `bower.json` file in the root of the project.

File: web-app/js/source/topicDialog.js [hunk 3]
```diff
@@ -532,9 +532,55 @@ function deleteResource (resourceIDs) {
...
+			}
+			if ($(this).val() === 'Very Important') {
+				numberOfVeryImportantTopics += 1;
+			}
+			if ($(this).val() === 'Good to Know') {
+				numberOfGoodToKnowTopics += 1;
+			}
+		});
+		$('#chart').chart({
+			data: new Array(numberOfCriticalTopics, numberOfGoodToKnowTopics, numberOfVeryImportantTopics),
```
Comment (ChristianMurphy) [inline web-app/js/source/topicDialog.js]: In JS the `Array` constructor takes in a single parameter `size`.
To initialize items in an `Array` use this instead
``` javascript
data: [numberOfCriticalTopics, numberOfGoodToKnowTopics, numberOfVeryImportantTopics]
```

**Comment by ChristianMurphy** [comment]
My comment from #478 still holds true
> We should probably stay away from [Canvas JS](http://canvasjs.com/).
> The [Attribution-NonCommercial 3.0 Unported license](https://creativecommons.org/licenses/by-nc/3.0/deed.en_US) requires buying a license for commercial use.
>
> While this is not a problem now, it could become a problem down the road.
>
> I would recommend looking into [C3 js](http://c3js.org/) or [Chart js](http://www.chartjs.org/) both of which are licensed under the more permissive [MIT license](http://opensource.org/licenses/MIT) which does not limit usage, or require buying a license.

It would be better to do a small rewrite now, than a large rewrite later.

**Comment by ChristianMurphy** [comment]
Also a note :notebook:
The [Travis CI](https://travis-ci.org/IMOD-ASU/imod) checks that show up on each pull request, are running the same build process that is used on the development and production server.  :construction_worker: 
So if an :x: pops up in the PR, please don't merge, the development server will also fail.

**Comment by kliyer** [comment]
Hey Christian,

The ESLINT is giving an error called variable name too small. In case of
the plugin (canvas.js) there is a variable y used which is inbuilt to the
plugin any changes to that variable does not render the graph. Also
although one of the functions in topicDialog.js is giving an error if the
function name starts with a capital letter it should be a constructor. The
function is a constructor in the plugin. We are using canvas.js as a
temporary solution i have a ticket assigned to look for a new one.
 If possible kindly let me know what could be done for the above problems.
 It would be really helpful.

Thanks and Regards
Kartik Iyer

**Comment by ChristianMurphy** [comment]
@kliyer I temporarily made the `new-cap` and `id-length` rules warning, instead of error.
That should resolve the above issues you mentioned.

**Comment by ChristianMurphy** [comment]
also since `canvas.js` does not appear to be on the bower registry and does not have a Github repo.
The way to add it to bower would be to find the download url, and use that instead of a version or repo in the `bower.json` file.
E.G.
``` js
"canvas.js": "https://example.com/script.js"
```

**Comment by kliyer** [comment]
Hey Christian ,

Im sorry im bugging you with these issues. I tried finding a solution on
the internet read more about bower.json dependencies.

So far this is what i came up with.
"jquery.canvasjs.min.js": "http://canvasjs.com/fdm/chart/",
"jquery.piegraph.js": "
https://github.com/edgardleal/jquery.piegraph/archive/master.zip"

this is what the github documentation for bower.json says:Value must be a
valid semver range <https://github.com/npm/node-semver#ranges>, a Git URL,
or a URL (inc. tarball and zipball)

.Would this work fine ? Just do not want to run into issues again so
confirming. It would be of great help if you could verify this.

Thanks and Regards,
Kartik Iyer

**Comment by ChristianMurphy** [comment]
Try this url for canvas js
`https://canvasjs.com/assets/script/canvasjs.min.js`
And this Github link for Piegraph JS
`edgardleal/jquery.piegraph`

**Pull request merged**